subsys/logging/log_backend_fs.c

```diff
@@ -339,14 +340,31 @@ static int allocate_new_file(struct fs_file_t *file)
 
 		curr_file_num = newest;
 
-		if (file_ctr >= 1) {
-			curr_file_num++;
-			if (curr_file_num > MAX_FILE_NUMERAL) {
-				curr_file_num = 0;
+		/* Is there space left in the newest file? */
+		snprintf(fname, sizeof(fname), "%s/%s%04d", CONFIG_LOG_BACKEND_FS_DIR,
+			 CONFIG_LOG_BACKEND_FS_FILE_PREFIX, curr_file_num);
+		rc = fs_open(file, fname, FS_O_CREATE | FS_O_WRITE | FS_O_APPEND);
+		if (rc < 0) {
+			goto out;
+		}
+		file_size = fs_tell(file);
+		if (file_size < CONFIG_LOG_BACKEND_FS_FILE_SIZE) {
```

Comment:
This should be comparison against some selected threshold, because there is no point to use the file if you can only fit, for example, one byte in it, because it will be immediately closed and new file will be allocated.
Also this code skips the logic that cleans out old logs in case where there is no space left on device (lines 383-401), and you may have still space left within file but you will  not be able to write anything anyway, before remove something or decide to overwrite some other file.

Reply:
Actually I don't think a threshold is needed.
This code only runs once when the backend is initialized, thus skipping the file cleanup at this time is ok because the code will have never let extra files be created during previous runs.

Also, see lines [154-177](https://github.com/zephyrproject-rtos/zephyr/pull/49700/files#diff-aef25f582f45618fe5f01cd4b464d21a31965570d559356fd99ea5ab2d1c11bfR154-R177).
`allocate_new_file` is called twice. Once for the initial init which will run this code to skip deleting the oldest file, then again if the newest file isn't large enough for the write. At that point it will clean up extra files if needed. This addresses your points.

Also [this test](https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/logging/log_backend_fs/src/log_fs_test.c#L189) passed CI proving extra files are handled properly.

Reply:
@de-nordic  Can you you re-review? The code already handles your concerns.